hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestOmDBInsightEndPoint.java

```diff
@@ -0,0 +1,329 @@
+        pipeline);
+    omKeyLocationInfoList2.add(omKeyLocationInfo5);
+
+    BlockID blockID6 = new BlockID(2, 3);
+    OmKeyLocationInfo omKeyLocationInfo6 = getOmKeyLocationInfo(blockID6,
+        pipeline);
+    omKeyLocationInfoList2.add(omKeyLocationInfo6);
+
+    OmKeyLocationInfoGroup omKeyLocationInfoGroup2 = new
+        OmKeyLocationInfoGroup(0, omKeyLocationInfoList2);
+
+    //key = key_three, Blocks = [ {CID = 2, LID = 2}, {CID = 2, LID = 3} ]
+    writeDataToOm(reconOMMetadataManager,
+        "key_three", "bucketOne", "sampleVol",
+        Collections.singletonList(omKeyLocationInfoGroup2));
+
+    //Generate Recon container DB data.
+    OMMetadataManager omMetadataManagerMock = mock(OMMetadataManager.class);
+    Table tableMock = mock(Table.class);
+    when(tableMock.getName()).thenReturn("KeyTable");
+    when(omMetadataManagerMock.getKeyTable(getBucketLayout()))
+        .thenReturn(tableMock);
+    ContainerKeyMapperTask containerKeyMapperTask  =
+        new ContainerKeyMapperTask(reconContainerMetadataManager);
+    containerKeyMapperTask.reprocess(reconOMMetadataManager);
+  }
+
+  @Test
+  public void testGetOpenKeyInfo() throws Exception {
+    OmKeyInfo omKeyInfo = getOmKeyInfo("sampleVol", "bucketOne", "key_one");
+
+    reconOMMetadataManager.getOpenKeyTable(getBucketLayout())
+        .put("/sampleVol/bucketOne/key_one", omKeyInfo);
+    OmKeyInfo omKeyInfo1 =
+        reconOMMetadataManager.getOpenKeyTable(getBucketLayout())
+            .get("/sampleVol/bucketOne/key_one");
+    Assertions.assertEquals("key_one", omKeyInfo1.getKeyName());
+    Response openKeyInfoResp = omdbInsightEndpoint.getOpenKeyInfo(-1, "");
+    KeyInsightInfoResp keyInsightInfoResp =
+        (KeyInsightInfoResp) openKeyInfoResp.getEntity();
+    Assertions.assertNotNull(keyInsightInfoResp);
+    Assertions.assertEquals("key_one",
+        keyInsightInfoResp.getNonFSOKeyInfoList().get(0).getPath());
+  }
+
+  @Test
+  public void testGetDeletedKeyInfo() throws Exception {
+    OmKeyInfo omKeyInfo = getOmKeyInfo("sampleVol", "bucketOne", "key_one");
+
+    reconOMMetadataManager.getKeyTable(getBucketLayout())
+        .put("/sampleVol/bucketOne/key_one", omKeyInfo);
+    OmKeyInfo omKeyInfo1 = reconOMMetadataManager.getKeyTable(getBucketLayout())
+        .get("/sampleVol/bucketOne/key_one");
+    Assertions.assertEquals("key_one", omKeyInfo1.getKeyName());
+    RepeatedOmKeyInfo repeatedOmKeyInfo = new RepeatedOmKeyInfo(omKeyInfo);
+    reconOMMetadataManager.getDeletedTable()
+        .put("/sampleVol/bucketOne/key_one", repeatedOmKeyInfo);
+    RepeatedOmKeyInfo repeatedOmKeyInfo1 =
+        reconOMMetadataManager.getDeletedTable()
+            .get("/sampleVol/bucketOne/key_one");
+    Assertions.assertEquals("key_one",
+        repeatedOmKeyInfo1.getOmKeyInfoList().get(0).getKeyName());
+    Response deletedKeyInfo = omdbInsightEndpoint.getDeletedKeyInfo(-1, "");
```

Comment:
Currently, the `testGetDeletedKeyInfo()` method is only testing a single case. It would be good to test multiple cases such as when there are **multiple open keys**  in the open key table or when the `limit` and `prevKeyPrefix` query parameters are used.

Reply:
@ArafatKhan2198  - thanks for review. I have added few more test cases.